src/packages/cli/src/__tests__/__helpers__/integrationTest.ts

```diff
@@ -255,15 +262,114 @@ export function integrationTest<Client>(input: Input<Client>) {
       await state.prisma.$connect()
 
       const result = await scenario.do(state.prisma)
-
       expect(result).toEqual(scenario.expect)
-      expect(prepareSchemaForSnapshot(datamodel)).toMatchSnapshot(`datamodel`)
-      expect(introspectionResult.warnings).toMatchSnapshot(`warnings`)
+
+      await teardownScenario(state)
     },
-    input.settings?.timeout ?? 15_000,
+    input.settings?.timeout ?? 30_000,
   )
 }
 
+async function afterAllScenarios(
+  kind: string,
+  states: Record<string, ScenarioState>,
+) {
+  engine.stop()
+  Object.entries(states).forEach(async ([_, state]) => {
+    // props might be missing if test errors out before they are set.
+    if (state.db && state.input.database.close) {
+      await state.input.database.close(state.db)
+    }
+    if (state.input.settings?.cleanupTempDirs !== false) {
+      fs.remove(getScenariosDir(state.input.database.name, kind))
+    }
+  })
+}
+
+function beforeAllScenarios(testKind: string, input: Input) {
+  fs.remove(getScenariosDir(input.database.name, testKind))
+}
+
+/**
+ * Setup the scenario test context and state and Prisma datamodel.
+ */
+async function setupScenario(kind: string, input: Input, scenario: Scenario) {
+  const state: ScenarioState = {} as any
+  const ctx: Context = {} as any
+  ctx.fs = fs.cwd(getScenarioDir(input.database.name, kind, scenario.name))
+  ctx.scenarioName = `${kind}: ${scenario.name}`
+  ctx.scenarioSlug = slugify(ctx.scenarioName, { separator: '_' })
+  ctx.id = `${ctx.scenarioSlug.slice(0, 7)}_${hash(ctx.scenarioSlug)}`
```

Comment:
We cannot simply use `scenarioSlug` because its length exceeds identifier limits in some databases (e.g. Postgresql).